### PR TITLE
Use -Werror when building in CI

### PR DESCRIPTION
**Pull request by ret2libc**
**Closing issues**

Closes https://github.com/radareorg/radare2/issues/17780

### CODE REVIEW

**Comment by trufae** [comment]
Will you fix the current warnings in this PR before merging it?

**Comment by ret2libc** [comment]
> Will you fix the current warnings in this PR before merging it?

Yes sure, otherwise it would be red. Done.

**Comment by XVilka** [comment]
```
p/../arch/8051/8051_disas.c:64:14: error: ‘r_8051_disas’ defined but not used [-Werror=unused-function]
 static char *r_8051_disas(ut64 pc, const ut8 *buf, int len, int *olen) {
              ^~~~~~~~~~~~
cc1: all warnings being treated as errors
make[3]: *** [p/../arch/8051/8051_disas.o] Error 1
```

**Comment by XVilka** [comment]
Should fix the capstone compilation error: https://github.com/aquynh/capstone/pull/1695

**Comment by ret2libc** [comment]
> Should fix the capstone compilation error: [aquynh/capstone#1695](https://github.com/aquynh/capstone/pull/1695)

Until that is not merged, I removed the -Werror from acr builds (as those pass the cflags to capstone as well) so we can merge this and extend it once capstone fix lands upstream.

**Review by XVilka — APPROVED** [review]

**Comment by XVilka** [comment]
Unrelated to this PR, but some useful warnings from MSVC:
```
c:\projects\radare2\libr\asm\arch\xtensa\gnu\xtensa-modules.c(4130): warning C4146: unary minus operator applied to unsigned type, result still unsigned [C:\projects\radare2\build\libr\anal\04634f8@@r_anal@sta.vcxproj]
c:\projects\radare2\libr\debug\p\debug_gdb.c(539): warning C4133: 'initializing': incompatible types - from 'RDebugReasonType (__cdecl *)(RDebug *,int)' to 'int (__cdecl *)(RDebug *,int)' [C:\projects\radare2\build\libr\debug\e931636@@r_debug@sta.vcxproj]
c:\projects\radare2\libr\debug\p\debug_native.c(1596): warning C4133: 'initializing': incompatible types - from 'RDebugReasonType (__cdecl *)(RDebug *,int)' to 'int (__cdecl *)(RDebug *,int)' [C:\projects\radare2\build\libr\debug\e931636@@r_debug@sta.vcxproj]
c:\projects\radare2\libr\debug\p\debug_winkd.c(268): warning C4133: 'initializing': incompatible types - from 'RDebugReasonType (__cdecl *)(RDebug *,int)' to 'int (__cdecl *)(RDebug *,int)' [C:\projects\radare2\build\libr\debug\e931636@@r_debug@sta.vcxproj]
cl : Command line warning D9002: ignoring unknown option '-std=c99' [C:\projects\radare2\build\libr\core\f105261@@r_core@sta.vcxproj]
cl : Command line warning D9002: ignoring unknown option '-std=c99' [C:\projects\radare2\build\libr\core\f105261@@r_core@sta.vcxproj]
cl : Command line warning D9002: ignoring unknown option '-std=c99' [C:\projects\radare2\build\libr\core\f105261@@r_core@sta.vcxproj]
cl : Command line warning D9002: ignoring unknown option '-std=c99' [C:\projects\radare2\build\libr\core\f105261@@r_core@sta.vcxproj]
cl : Command line warning D9002: ignoring unknown option '-std=c99' [C:\projects\radare2\build\libr\core\f105261@@r_core@sta.vcxproj]
cl : Command line warning D9002: ignoring unknown option '-std=c99' [C:\projects\radare2\build\libr\core\f105261@@r_core@sta.vcxproj]
cl : Command line warning D9002: ignoring unknown option '-std=c99' [C:\projects\radare2\build\libr\core\f105261@@r_core@sta.vcxproj]
cl : Command line warning D9002: ignoring unknown option '-std=c99' [C:\projects\radare2\build\libr\core\f105261@@r_core@sta.vcxproj]
```

**Comment by XVilka** [comment]
And one in WinDbg code cc @GustavoLCR @yossizap 
```
../libr/debug/p/debug_windbg.c(74,10): warning: implicit declaration of function 'windbg_wait' is invalid in C99 [-Wimplicit-function-declaration]
                return windbg_wait (dbg, dbg->pid) != R_DEBUG_REASON_ERROR;
                       ^
../libr/debug/p/debug_windbg.c(82,25): warning: unused variable 'Qualifier' [-Wunused-variable]
        ULONG Id = tid, Class, Qualifier;
                               ^
../libr/debug/p/debug_windbg.c(82,18): warning: unused variable 'Class' [-Wunused-variable]
        ULONG Id = tid, Class, Qualifier;
                        ^
../libr/debug/p/debug_windbg.c(165,52): warning: incompatible pointer types passing 'int *' to parameter of type 'PULONG' (aka 'unsigned long *') [-Wincompatible-pointer-types]
                ITHISCALL (dbgSysObj, GetCurrentProcessSystemId, &dbg->pid);
                                                                 ^~~~~~~~~
../libr/debug/p/debug_windbg.c(27,88): note: expanded from macro 'ITHISCALL'
#define ITHISCALL(dbginterface, function, ...) THISCALL (idbg->dbginterface, function, __VA_ARGS__)
                                                                                       ^~~~~~~~~~~
../libr/debug/p/debug_windbg.c(26,93): note: expanded from macro 'THISCALL'
#define THISCALL(dbginterface, function, ...) dbginterface->lpVtbl->function (dbginterface, __VA_ARGS__)
                                                                                            ^~~~~~~~~~~
../libr/debug/p/debug_windbg.c(166,51): warning: incompatible pointer types passing 'int *' to parameter of type 'PULONG' (aka 'unsigned long *') [-Wincompatible-pointer-types]
                ITHISCALL (dbgSysObj, GetCurrentThreadSystemId, &dbg->tid);
                                                                ^~~~~~~~~
../libr/debug/p/debug_windbg.c(27,88): note: expanded from macro 'ITHISCALL'
#define ITHISCALL(dbginterface, function, ...) THISCALL (idbg->dbginterface, function, __VA_ARGS__)
                                                                                       ^~~~~~~~~~~
../libr/debug/p/debug_windbg.c(26,93): note: expanded from macro 'THISCALL'
#define THISCALL(dbginterface, function, ...) dbginterface->lpVtbl->function (dbginterface, __VA_ARGS__)
                                                                                            ^~~~~~~~~~~
../libr/debug/p/debug_windbg.c(220,40): warning: incompatible pointer types passing 'int *' to parameter of type 'volatile long *' [-Wincompatible-pointer-types]
                        b->internal = InterlockedIncrement (&bp_idx);
                                                            ^~~~~~~
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\winnt.h(3019,50): note: passing argument to parameter 'Addend' here
    _Inout_ _Interlocked_operand_ LONG volatile *Addend
                                                 ^
../libr/debug/p/debug_windbg.c(248,39): warning: incompatible pointer types passing 'int *' to parameter of type 'PULONG' (aka 'unsigned long *') [-Wincompatible-pointer-types]
        THISCALL (bkpt, GetCurrentPassCount, &b->togglehits);
                                             ^~~~~~~~~~~~~~
../libr/debug/p/debug_windbg.c(26,93): note: expanded from macro 'THISCALL'
#define THISCALL(dbginterface, function, ...) dbginterface->lpVtbl->function (dbginterface, __VA_ARGS__)
                                                                                            ^~~~~~~~~~~
../libr/debug/p/debug_windbg.c(279,89): warning: incompatible pointer types passing 'int *' to parameter of type 'PULONG' (aka 'unsigned long *') [-Wincompatible-pointer-types]
        if (!idbg || !idbg->initialized || FAILED (ITHISCALL (dbgCtrl, GetActualProcessorType, &type))) {
                                                                                               ^~~~~
../libr/debug/p/debug_windbg.c(27,88): note: expanded from macro 'ITHISCALL'
#define ITHISCALL(dbginterface, function, ...) THISCALL (idbg->dbginterface, function, __VA_ARGS__)
                                                                                       ^~~~~~~~~~~
../libr/debug/p/debug_windbg.c(26,93): note: expanded from macro 'THISCALL'
#define THISCALL(dbginterface, function, ...) dbginterface->lpVtbl->function (dbginterface, __VA_ARGS__)
                                                                                            ^~~~~~~~~~~
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\shared\winerror.h(28988,32): note: expanded from macro 'FAILED'
#define FAILED(hr) (((HRESULT)(hr)) < 0)
                               ^~
8 warnings generated.
[664/1543] Compiling C object libr/debug/r_debug.dll.p/p_native_maps_windows_maps.c.obj
[665/1543] Compiling C object libr/core/r_core.dll.p/casm.c.obj
[666/1543] Compiling C object libr/debug/r_debug.dll.p/p_debug_winkd.c.obj
[667/1543] Compiling C object libr/debug/r_debug.dll.p/p_native_windows_windows_debug.c.obj
../libr/debug/p/native/windows/windows_debug.c(1482,34): warning: cast to 'HANDLE' (aka 'void *') from smaller integer type 'USHORT' (aka 'unsigned short') [-Wint-to-void-pointer-cast]
                if (w32_NtDuplicateObject (ph, (HANDLE)handle.Handle, GetCurrentProcess (), &dupHandle, 0, 0, 0)) {
                                               ^
1 warning generated.
```

File: libr/socket/run.c
Comment (wargio) [inline libr/socket/run.c:289]: are we sure that this does not break anything?

Comment (ret2libc) [reply libr/socket/run.c:289]: yes, there were warnings in macos about not used `handle_redirection_proc` and `restore_saved_fd`. Those functions are used only in `handle_redirection` and `handle_redirection` is just a `return 0;` in this case, see https://github.com/radareorg/radare2/pull/17781/files#diff-91ebf3f9d4c19931970b6445d178761e435da9628a060c655969519191eee820R390

Comment (wargio) [reply libr/socket/run.c:289]: okok. good.

**Comment by trufae** [comment]
🎉

**Comment by XVilka** [comment]
It broke Capstone 3 builds. Maybe we should just silently skip instead of emitting a `#warning`?
```
FAILED: libr/asm/8a8d21f@@r_asm@sha/p_asm_m68k_cs.c.o 
gcc -Ilibr/asm/8a8d21f@@r_asm@sha -I. -I../ -I../libr/include -Ilibr/asm/../../shlr -I../libr/asm/../../shlr -I../libr/asm/arch/include -I../libr/asm/arch -I../libr/asm/arch/h8300 -I../libr/asm/arch/hexagon -I../libr/asm/arch/msp430 -I../libr/asm/arch/rsp -I../libr/asm/arch/mcore -I../libr/asm/arch/v850 -I../libr/asm/arch/propeller -I../libr/asm/arch/ebc -I../libr/asm/arch/cr16 -I../libr/asm/arch/8051 -I../libr/asm/arch/v810 -I../libr/asm/arch/or1k -I../shlr/spp -I../shlr/sdb/src -I../shlr/zip/zlib -I../shlr/capstone/include -I../shlr/java -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -g -fvisibility=hidden -Werror -fPIC -pthread -DR2_PLUGIN_INCORE=1 -MD -MQ 'libr/asm/8a8d21f@@r_asm@sha/p_asm_m68k_cs.c.o' -MF 'libr/asm/8a8d21f@@r_asm@sha/p_asm_m68k_cs.c.o.d' -o 'libr/asm/8a8d21f@@r_asm@sha/p_asm_m68k_cs.c.o' -c ../libr/asm/p/asm_m68k_cs.c
../libr/asm/p/asm_m68k_cs.c:14:2: error: #warning Cannot find capstone-m68k support [-Werror=cpp]
 #warning Cannot find capstone-m68k support
  ^~~~~~~
cc1: all warnings being treated as errors
[733/1528] Compiling C object 'libr/asm/8
``

**Comment by ret2libc** [comment]
We could try `-Wno-cpp  (C, Objective-C, C++, Objective-C++ and Fortran only) Suppress warning messages emitted by #warning directives.` (according to GCC pages). I will do this in a separate pr.

**Pull request merged**
